### ISSUE TITLE
Metadata with empty value is dropped.

### ISSUE THREAD

**Issue by JenSte** [tus/tusd#397]
**Describe the bug**

The protocol definition [states](https://tus.io/protocols/resumable-upload.html#upload-metadata) that metadata values may be left out: "[...] The value MAY be empty. In these cases, the space, which would normally separate the key and the value, MAY be left out."

However it seems that tusd is dropping such metadata without a value.

**To Reproduce**
1. Upload a file to the tus.io demo server (I assume the demo server is using tusd(?), and the same behavior is observable if I start tusd locally.)
```
$ aiotus-upload http://master.tus.io/files test.bin --metadata is_confidential --metadata key=value
http://master.tus.io/files/0117a6f71...
```
The captured HTTP traffic of the upload creation is shown below. Note that the `Upload-Metadata` header contains three metadata items with a value ("filename", "mime_type", and "key") and one item without a key ("is_confidential").
```
POST /files HTTP/1.1
Host: master.tus.io
Tus-Resumable: 1.0.0
Upload-Length: 10
Upload-Metadata: filename dGVzdC5iaW4=, mime_type YXBwbGljYXRpb24vb2N0ZXQtc3RyZWFt, is_confidential, key dmFsdWU=
Accept: */*
Accept-Encoding: gzip, deflate
User-Agent: Python/3.8 aiohttp/3.6.2
Content-Length: 0
Content-Type: application/octet-stream

HTTP/1.1 201 Created
Server: openresty/1.15.8.1
Date: Tue, 16 Jun 2020 21:42:23 GMT
Content-Length: 0
Connection: keep-alive
Location: http://master.tus.io/files/0117a6f71a8c855138d548e7ef93d385+T.VAZpfSSs5WK19jNGy1eOugruRLoPNVGy4wNKJ8ud9HdCVxUXgBwH23xc0fIGslUvg7jnlpEckxKusOG.LJTSCiH2iJqlV9zW.0kECchfXLxjpkGChE4yQlwqaUvwD9
Tus-Resumable: 1.0.0
X-Content-Type-Options: nosniff

...
```
2. When asking the server for the metadata, the item without a value is not included in the response:
```
$ aiotus-metadata http://master.tus.io/files/0117a6f71...
key: value
mime_type: application/octet-stream
filename: test.bin
```
HTTP traffic:
```
HEAD /files/0117a6f71a8c855138d548e7ef93d385+T.VAZpfSSs5WK19jNGy1eOugruRLoPNVGy4wNKJ8ud9HdCVxUXgBwH23xc0fIGslUvg7jnlpEckxKusOG.LJTSCiH2iJqlV9zW.0kECchfXLxjpkGChE4yQlwqaUvwD9 HTTP/1.1
Host: master.tus.io
Tus-Resumable: 1.0.0
Accept: */*
Accept-Encoding: gzip, deflate
User-Agent: Python/3.8 aiohttp/3.6.2

HTTP/1.1 200 OK
Server: openresty/1.15.8.1
Date: Tue, 16 Jun 2020 22:02:13 GMT
Connection: keep-alive
Cache-Control: no-store
Tus-Resumable: 1.0.0
Upload-Length: 10
Upload-Metadata: key dmFsdWU=,mime_type YXBwbGljYXRpb24vb2N0ZXQtc3RyZWFt,filename dGVzdC5iaW4=
Upload-Offset: 10
X-Content-Type-Options: nosniff
```
3. When testing this with a tusd server running locally, it can be seen that the metadata item without the value is not stored in the .info file, while the other items are:
```
[...] "MetaData":{"filename":"test.bin","key":"value","mime_type":"application/octet-stream"} [...]
```

**Expected behavior**
The server should store and return metadata without values.

**Setup details**
Please provide following details, if applicable to your situation:
- Operating System: Linux
- Used tusd version: Whatever version is running on master.tus.io, or v1.3.0 locally
- Used tusd data storage: master.tus.io: ?, locally: Disk storage
- Used tusd configuration: None
- Used tus client library: [aiotus](https://github.com/JenSte/aiotus), "dev" branch that allows the demo CLI tool to upload metadata without values

**Comment by Acconut** [comment]
Thank you very much for the detailed report! You are correct in that tusd requires a value to be present, which is also mentioned in the code: https://github.com/tus/tusd/blob/c822a3afb1fcd644b161ecda8454c40236f4d8ad/pkg/handler/unrouted_handler.go#L1123-L1126

I am happy to accept a PR for a fix if you are interested :)

**Comment by JenSte** [comment]
The linked merge request maps empty metadata values to empty strings. With this change tusd also stores and returns the metadata keys without values.

Small nitpick: It is still not possible to distinguish the case where the value is completely omitted (```Upload-Metadata: key```) from the case where the value is an empty string in the beginning and also after base64-encoding it (```Upload-Metadata: key ```, note the space at the end of the header), in both cases the same result would be returned when asking for the metadata later. To make this distinction possible the type used to store the metadata would have to be changed from ```map[string]string``` to ```map[string]*string```. I tried this but had several problems:
- Changing the type would result in big changes in various places throughout the code of tusd.
- ```map[string]string``` is also used in the protobuf file for GRPC.
- Not all data stores support this. For example, S3 uses ```map[string]*string``` and would probably [allow it](https://godoc.org/github.com/aws/aws-sdk-go/service/s3#CreateMultipartUploadInput.SetMetadata), while GCS also [uses](https://godoc.org/github.com/tus/tusd/pkg/gcsstore#GCSService.SetObjectMetadata) ```map[string]string``` and does not allow to omit values completely.

**Comment by Acconut** [comment]
Apologies for the delay in my side! I will have a look at your PR in the next few days.

**Comment by Acconut** [comment]
Fixed in #400.
